### ISSUE TITLE
Bazel build broken on MacOS/Darwin

### ISSUE THREAD

**Issue by aohren** [grpc/grpc#23096]
### What version of gRPC and what language are you using?
* grpc/grpc 1.29.1 (7d89dbb311f049b43bda7bbf6f7d7bf1b4c24419) 
* bazel 3.1.0

### What operating system (Linux, Windows,...) and version?
* MacOS 10.15.5 (`19.5.0 Darwin Kernel Version 19.5.0: Thu Apr 30 18:25:59 PDT 2020; root:xnu-6153.121.1~7/RELEASE_X86_64 x86_64`)

### What runtime / compiler are you using (e.g. python version or version of gcc)
```
aohren-macbookpro2% python --version 
Python 2.7.16
aohren-macbookpro2% python3 --version
Python 3.6.8
aohren-macbookpro2% gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.0 (clang-1100.0.33.17)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

### What did you do?
I attempted to build the `@com_github_grpc_grpc//:grpc` Bazel target (and various individual targets that depend on grpc) in the [googleapis/googleapis](https://github.com/googleapis/googleapis) workspace on Darwin. 

### What did you expect to see?
A successful build.

### What did you see instead?
A link error (likely the same as #13856, but it's marked as resolved). This reproduces for other targets within googleapis that depend on grpc, preventing ability to build pretty much any googlepais project on MacOS.  

```
aohren-macbookpro2% bazel build --verbose_failures --sandbox_debug @com_github_grpc_grpc//:grpc
DEBUG: Rule 'bazel_skylib' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "1dde365491125a3db70731e25658dfdd3bc5dbdfd11b840b3e987ecf043c7ca0"
DEBUG: Repository bazel_skylib instantiated at:
  no stack (--record_rule_instantiation_callstack not enabled)
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_aohren/30304f4e587294bcdc36fbba9810c371/external/bazel_tools/tools/build_defs/repo/http.bzl:336:16: in <toplevel>
INFO: Analyzed target @com_github_grpc_grpc//:grpc (3 packages loaded, 154 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_aohren/30304f4e587294bcdc36fbba9810c371/external/com_github_grpc_grpc/BUILD:330:1: Linking of rule '@com_github_grpc_grpc//:grpc' failed (Exit 1): sandbox-exec failed: error executing command 
  (cd /private/var/tmp/_bazel_aohren/30304f4e587294bcdc36fbba9810c371/sandbox/darwin-sandbox/864/execroot/com_google_googleapis && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=10.15 \
    DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer \
    PATH='/Users/aohren/go/bin:/opt/local/bin:/opt/local/sbin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Applications/Wireshark.app/Contents/MacOS' \
    SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk \
    TMPDIR=/var/folders/f0/s1hltlwx4g19ywmd6cf3sdpc004rbz/T/ \
    XCODE_VERSION_OVERRIDE=11.3.1.11C505 \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_aohren/30304f4e587294bcdc36fbba9810c371/sandbox/darwin-sandbox/864/sandbox.sb /var/tmp/_bazel_aohren/install/8e88328f69e8201e7952615a49646cb0/process-wrapper '--timeout=0' '--kill_delay=15' external/local_config_cc/cc_wrapper.sh -lc++ -fobjc-link-runtime -Wl,-S -shared -o bazel-out/darwin-fastbuild/bin/external/com_github_grpc_grpc/libgrpc.so bazel-out/darwin-fastbuild/bin/external/com_github_grpc_grpc/_objs/grpc/init.o bazel-out/darwin-fastbuild/bin/external/com_github_grpc_grpc/_objs/grpc/grpc_plugin_registry.o -headerpad_max_install_names -pthread -no-canonical-prefixes -undefined dynamic_lookup '-mmacosx-version-min=10.15') sandbox-exec failed: error executing command 
  (cd /private/var/tmp/_bazel_aohren/30304f4e587294bcdc36fbba9810c371/sandbox/darwin-sandbox/864/execroot/com_google_googleapis && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=10.15 \
    DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer \
    PATH='/Users/aohren/go/bin:/opt/local/bin:/opt/local/sbin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Applications/Wireshark.app/Contents/MacOS' \
    SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk \
    TMPDIR=/var/folders/f0/s1hltlwx4g19ywmd6cf3sdpc004rbz/T/ \
    XCODE_VERSION_OVERRIDE=11.3.1.11C505 \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_aohren/30304f4e587294bcdc36fbba9810c371/sandbox/darwin-sandbox/864/sandbox.sb /var/tmp/_bazel_aohren/install/8e88328f69e8201e7952615a49646cb0/process-wrapper '--timeout=0' '--kill_delay=15' external/local_config_cc/cc_wrapper.sh -lc++ -fobjc-link-runtime -Wl,-S -shared -o bazel-out/darwin-fastbuild/bin/external/com_github_grpc_grpc/libgrpc.so bazel-out/darwin-fastbuild/bin/external/com_github_grpc_grpc/_objs/grpc/init.o bazel-out/darwin-fastbuild/bin/external/com_github_grpc_grpc/_objs/grpc/grpc_plugin_registry.o -headerpad_max_install_names -pthread -no-canonical-prefixes -undefined dynamic_lookup '-mmacosx-version-min=10.15')
ld: illegal thread local variable reference to regular symbol __ZN9grpc_core7ExecCtx9exec_ctx_E for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Target @com_github_grpc_grpc//:grpc failed to build
INFO: Elapsed time: 6.701s, Critical Path: 1.17s
INFO: 28 processes: 28 darwin-sandbox.
FAILED: Build did NOT complete successfully
```
### Anything else we should know about your project / environment?

Steps to reproduce on Darwin:
* `git clone git@github.com:googleapis/googleapis.git`
* `cd googleapis`
* `bazel build --sandbox_debug @com_github_grpc_grpc//:grpc`

The googleapis [WORKSPACE](https://github.com/googleapis/googleapis/blob/master/WORKSPACE) currently pulls in 1.25.0 [8347f4753568b5b66e49111c60ae2841278d3f33 ] of grpc, but the error reproduces even if I update to latest tagged release with these edits before building per above:

```
aohren-macbookpro2% git diff
diff --git a/WORKSPACE b/WORKSPACE
index a30daf2a7..c92ccc94b 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -111,14 +111,18 @@ go_register_toolchains()
 
 http_archive(
     name = "com_github_grpc_grpc",
-    strip_prefix = "grpc-8347f4753568b5b66e49111c60ae2841278d3f33",  # this is 1.25.0 with fixes
-    urls = ["https://github.com/grpc/grpc/archive/8347f4753568b5b66e49111c60ae2841278d3f33.zip"],
+    strip_prefix = "grpc-7d89dbb311f049b43bda7bbf6f7d7bf1b4c24419",  # this is 1.29.1
+    urls = ["https://github.com/grpc/grpc/archive/7d89dbb311f049b43bda7bbf6f7d7bf1b4c24419.zip"],
 )
 
 load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
 
 grpc_deps()
 
+load("@com_github_grpc_grpc//bazel:grpc_extra_deps.bzl", "grpc_extra_deps")
+
+grpc_extra_deps()
+
 load("@upb//bazel:repository_defs.bzl", "bazel_version_repository")
 
 bazel_version_repository(
```

**Comment by markdroth** [comment]
Closing this as a dup of #13856.

**Comment by benh** [comment]
For others that might run into this, I ran into this only when using `cc_test` (not `cc_binary`). Adding `linkstatic = True` to my `cc_test` resolved this problem for me (for now while I wait for an official fix from #13856).

**Comment by ipodfans** [comment]
@benh Thanks Ben. I can confirm that the problem goes away when add "linkstatic = 1" to cc_test.
